### PR TITLE
Feat/trigger button

### PR DESCRIPTION
**Pull request by devceline**
# Changes
- Add licenses
- Fix build chain
- Add proper dev scripts 
- Add trigger button
- Using RxJS for minimal eventful state

### CODE REVIEW

File: packages/react/package.json [hunk 1]
```diff
@@ -9,12 +9,20 @@
   "files": [
     "dist"
   ],
-  "license": "MIT",
+  "license": "Apache-2.0",
   "scripts": {
-    "build": "yarn build:cjs &&yarn build:esm && yarn build:types",
+    "build": "yarn build:clean && yarn build:cjs &&yarn build:esm && yarn build:types",
```
Comment (Cali93) [inline packages/react/package.json:14]: I suggest you to look into [`tsup`](https://www.npmjs.com/package/tsup) for the build tool.
It'll make this more readable/maintainable.

File: packages/react/src/W3iWidget/index.tsx [hunk 2]
```diff
@@ -23,7 +24,7 @@ const htmlifyParams = (
 ): { [k: string]: number | string } => {
   return Object.fromEntries(
     Object.entries(params)
-      .filter(([_, v]) => typeof v !== "function")
+      .filter(([_, v]) => typeof v !== "function" && typeof v !== "object")
```
Comment (Cali93) [inline packages/react/src/W3iWidget/index.tsx:27]: Could you provide context with a small comment on what's going on here ?

File: packages/tsconfig/nextjs.json [hunk 3]
```diff
@@ -4,9 +4,11 @@
   "extends": "./base.json",
   "compilerOptions": {
     "plugins": [{ "name": "next" }],
+		"moduleResolution": "node",
```
Comment (Cali93) [inline packages/tsconfig/nextjs.json:7]: nit: indentation.
Wouldn't happen with VS Code 😆

File: packages/ui/src/components/w3i-widget/index.ts [hunk 4]
```diff
@@ -83,7 +84,31 @@ export class W3iWidget extends LitElement {
...
+    });
+
+    this.addEventListener("toggleWidget", () => {
+      if (!this.iframeRef.value) return;
+
+      const currentStyle = this.iframeRef.value.style.display;
+
+      const set = this.iframeRef.value.style.setProperty;
+
+      switch (currentStyle) {
```
Comment (Cali93) [inline packages/ui/src/components/w3i-widget/index.ts:104]: Should we add a default case ?

**Review by Cali93 — APPROVED** [review]
LGTM 👌 
Just minor comments on tooling.

File: packages/react/package.json (hunk 1 above)
Comment (devceline) [reply packages/react/package.json:14]: I really don't want to introduce more stuff right now, if we need to in the future I'll look into it. This is the setup on viem so I'm sure it's good enough for now

File: packages/ui/src/components/w3i-widget/index.ts (hunk 4 above)
Comment (devceline) [reply packages/ui/src/components/w3i-widget/index.ts]: This is no longer used, good catch

File: packages/tsconfig/nextjs.json (hunk 3 above)
Comment (devceline) [reply packages/tsconfig/nextjs.json]: really need to update my vim setup istg

**Pull request merged**